### PR TITLE
Make CSRFActions use GraphStage (rebased #6328)

### PR DESCRIPTION
**Pull request by wsargent**
This is a rebased version of https://github.com/playframework/playframework/pull/6328 -- the only change here is to move to `GraphStage` in the CSRFAction so it doesn't use a deprecated type.

### CODE REVIEW

**Comment by wsargent** [comment]
@naferx this fails builds even with rebase -- does this work on your machine?

**Comment by naferx** [comment]
@wsargent Nope, it fails

File: framework/src/play-filters-helpers/src/main/scala/play/filters/csrf/CSRFActions.scala [hunk 1]
```diff
@@ -276,93 +277,105 @@ class CSRFAction(
...
-        if (checkBody(buffer)) {
-          // Switch to continue, and push the buffer
-          continue = true
-          if (ctx.isHoldingDownstream) {
-            val toPush = buffer
-            buffer = null
-            ctx.pushAndPull(toPush)
+private class BodyHandler(config: CSRFConfig, checkBody: ByteString => Boolean) extends GraphStage[FlowShape[ByteString, ByteString]] {
+
+  val in = Inlet[ByteString]("in")
```
Comment (ktoso) [inline framework/src/play-filters-helpers/src/main/scala/play/filters/csrf/CSRFActions.scala]: Highly recommnd calling them `BodyHandler.in` `BodyHandler.out`, it's better for debugging "which port was incorrectly pulled etc"

Comment (marcospereira) [inline framework/src/play-filters-helpers/src/main/scala/play/filters/csrf/CSRFActions.scala]: Hum, there are other places where we aren't following this practice. I will submit another PR to change that places (but not here).

File: framework/src/play-filters-helpers/src/main/scala/play/filters/csrf/CSRFActions.scala [hunk 2]
```diff
@@ -8,8 +8,9 @@ import java.util.Locale
 import javax.inject.Inject
 
 import akka.stream.Materializer
```
Comment (PromanSEW) [inline framework/src/play-filters-helpers/src/main/scala/play/filters/csrf/CSRFActions.scala]: Already imported by line below?

File: framework/src/play-filters-helpers/src/main/scala/play/filters/csrf/CSRFActions.scala (hunk 1 above)
Comment (ktoso) [inline framework/src/play-filters-helpers/src/main/scala/play/filters/csrf/CSRFActions.scala]: Great thanks :)

**Comment by ktoso** [comment]
Should I help you finishing this? We want this merged to upgrade to Akka 2.5 and PR some cool stuff 😉

**Comment by marcospereira** [comment]
@ktoso, yes please.

I think that if you want to cherry-pick the commit here and add it to your PR that would be ok too.

**Comment by ktoso** [comment]
FYI: I've started rewriting it, will want to unblock us next week with this and start PRing additional changes.

**Comment by ktoso** [comment]
This PR is superseded by https://github.com/playframework/playframework/pull/7153 
please close this one

**Pull request closed without merging**